content/en/docs/features/monitoring-and-diagnostics/monitoring-device-connectivity.md

```diff
@@ -0,0 +1,32 @@
+---
+title: 'Monitoring Device Connectivity'
+description: 'Learn how to determine the online or offline status of connected devices'
+date: '2023-08-08'
+categories: [features,iot-connectivity]
+keywords: [service, coap-gateway, network-monitoring]
+weight: 2
+---
+
+The CoAP gateway is responsible for the connection management including the up to date representation of the device status. This information is driven through the Resource Aggregate command - `UpdateDeviceMetadata`, that gets initiated when device connection status changes - device connects or disconnects from the CoAP Gateway. However, there are various scenarios to consider, such as unexpected termination of the CoAP Gateway, due to factors like node failure, power loss or others.
+
+## Managing Service Status Updates
+
+Every instance of the CoAP gateway is assigned a unique ID during initialization. These IDs are not only stored in a database but are also linked to device metadata to facilitate online state updates.
+
+At regular intervals, each CoAP gateway instance collaboratively updates a shared record containing their respective information, primarily their unique IDs. This update takes place periodically, typically once every minute. The specific record being updated changes over time, following a cyclic pattern (e.g., within a 1-minute period, each update corresponds to a different record). In the database, both the current shared record and the preceding one are retained.
+
+To calculate the appropriate `recordID` for updating the record within a 1-minute period, the following logic in Golang is employed:
+
+```golang
+    now := time.Now().UnixNano()
+    period := 1 * time.Minute
+    recordID = now - (now % int64(period))
+```
+
+Upon the generation of a new record, all records with a `recordID` corresponding to the current time minus twice the period are utilized to construct a merged list of all CoAP gateways that were operational before. Subsequently, CoAP gateways running within the time range of the current time and the current time minus twice the period are excluded from this list. Any remaining CoAP gateways in the list are deemed terminated, triggering the use of the Resource Aggregate command `TerminateConnectionsByService` to set associated devices with those CoAP gateways to an offline state. Subsequently, all the previous records that pertained to past operational periods are purged.
```

Comment:
It's almost impossible to understand this explanation by reading it. Could you please be more concrete with examples of merged list, concrete operations, and using more concrete values rather then "current time minus twice the period" mainly if you know that period is one minute?

And I am not sure with the solution in general, because if you have one coap gateway, which crashed, and cannot start again due to misconfiguration, or port unavailability, or not enough memory, devices will stay online. So many cases are not resolved, as you leave this responsibility to the quorum of CoAP Gateways.

Reply:
I created example where I explain it :)